src/chains/ethereum/ethereum/src/transaction-pool.ts

```diff
@@ -452,6 +453,40 @@ export default class TransactionPool extends Emittery<{ drain: undefined }> {
     }
     return null;
   }
+  /**
+   * Deep clones and resets the transaction pool's `executables` such that `inProgress`
+   * transactions are moved back to `pending` and all transactions are unlocked.
+   * @returns Cloned and reset copy of transaction pool's `executables`.
+   */
+  public cloneAndResetExecutables() {
+    const executables = cloneDeep(this.executables);
+    const { inProgress, pending } = executables;
+
+    if (pending.size > 0) {
```

Comment:
Same as below, I don't think you need this check

Reply:
Fixed!